### ISSUE TITLE
typevars becoming types with typevars doesn't work

### ISSUE THREAD

**Issue by ddfisher** [python/mypy#1236]
```python
from typing import TypeVar
T = TypeVar('T')
V = TypeVar('V')

def identity(x: T) -> T: ...
def type_var_using_fn(x: V) -> None: ...
identity(type_var_using_fn)
```
has the confusing error:
```
test.py:10: error: Argument 1 to "identity" has incompatible type Callable[[V], None]; expected Callable[[V], None]
```

This is most commonly seen when decorating functions with TypeVars.

**Comment by JukkaL** [comment]
This could be another manifestation of the old type variable renaming bug (#230 or #603 may be related.)

**Comment by rwbarton** [comment]
Here's what's happening here:

* We need to typecheck the expression `identity(type_var_using_fn)`.

* Since `identity` is a generic function, we have to infer the correct `T` to pick for the call based on the argument. mypy correctly determines that ``T = def [V] (x: V`-1)``, the type of the generic function `type_var_using_fn`.

* Having chosen `T`, we need to check that the arguments are actually valid. In this case, this means checking that the actual argument type ``def [V] (x: V`-1)`` is a subtype of the formal argument type ``T = def [V] (x: V`-1)``. Of course it is, but [checking whether a callable is a subtype of a generic callable is not yet supported by mypy](https://github.com/python/mypy/blob/dda38dca533ba323c9f9013291cb9c2524b2615f/mypy/subtypes.py#L205), resulting in the confusing error message.

I think the mypythonic thing to do here is to just implement this subtyping check: A callable is a subtype of a generic callable if it is so after the generic callable is instantiated at fresh ("skolem") types.

**Comment by ddfisher** [comment]
Thanks for tracking this down!  I agree, implementing that subtyping check sounds like the right fix here.

**Comment by gvanrossum** [comment]
Thanks! That sounds like a great analysis. Can you pull of the fix without
more help?

**Comment by rwbarton** [comment]
I have what should be a fix, but encountered #1335 while testing it.
